File: models/Muscle_Area.py
import json
class Muscle_Area:
    def __init__(self, id, name, muscles):
        self.id = id
        self.name = name
        self.muscles = muscles

    def set_id(self, id):
        self.id = id
    def set_name(self,name):
        self.name = name
    def set_muscles(self, muscles):
        self.muscles = muscles

    def get_id(self):
        return self.id
    def get_name(self):
        return self.name
    def get_muscles(self):
        return self.muscles
      

    def get_Muscle_data(self):
        area_data = {
            "id": self.id,
            "name": self.name,
        }
        return json.dumps(area_data)

    @staticmethod  
    def get_All_Area():
        pass

File: view/lib/models/Muscle_Area.dart
import 'dart:convert';

class MuscleArea {
  String id;
  String name;
  List<String> muscles;

  MuscleArea({
    required this.id,
    required this.name,
    required this.muscles,
  });

  // 靜態方法來模擬獲取所有肌肉區域
  static Future<List<MuscleArea>> getAllArea() async {
    // 實現你的邏輯，例如從API或數據庫獲取數據
    // 這裡我們返回一個空列表作為佔位符
    return [];
  }

  // 獲取肌肉區域數據
  String getMuscleData() {
    Map<String, dynamic> areaData = {
      "id": id,
      "name": name,
      "muscles": muscles,
    };
    return jsonEncode(areaData);
  }
}
